### PR TITLE
[Container] Add streaming commands

### PR DESCRIPTION
**Pull request by yolo3301**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

**Not really sure how to add tests for streaming commands. Please advise.**

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5405
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py [hunk 1]
```diff
@@ -57,4 +58,8 @@ def load_arguments(self, _):
         c.argument('azure_file_volume_mount_path', validator=validate_volume_mount_path, help='The path within the container where the volume should be mounted. Must not contain colon (:).')
 
     with self.argument_context('container logs') as c:
-        c.argument('container_name', help='The container name to tail the logs')
+        c.argument('container_name', help='The container name to tail the logs. If omitted, the first container in the container group will be chosen')
+        c.argument('follow', help='Indicate to stream the tailing logs', arg_type=get_three_state_flag(return_label=True))
```
Comment (williexu) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: We usually reserve the use of `get_three_state_flag` for registering boolean properties. For switches on the command, we recommend the `action='store_true'` kwarg.

File: src/command_modules/azure-cli-container/setup.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
     'azure-mgmt-containerinstance==0.2.0',
     'azure-cli-core',
     'pyyaml',
+    'colorama==0.3.9'
```
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/setup.py]: Verified both.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py [hunk 3]
```diff
@@ -166,15 +170,160 @@ def create_azure_file_volume_mount(azure_file_volume, azure_file_volume_mount_pa
...
+
+    try:
+        t = threading.Thread(target=stream_target, args=stream_args)
+        t.daemon = True
+        t.start()
+
+        while not terminate_condition(*terminate_condition_args) and t.is_alive():
+            time.sleep(10)
+
+        time.sleep(shupdown_grace_period)
```
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py]: Yes. Daemon will terminate also.

File: src/command_modules/azure-cli-container/README.rst [hunk 4]
```diff
@@ -177,3 +178,30 @@ Commands to list Azure container groups by resource group
...
+Commands to attach to a container in a container group
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+::
+
+    Command
+        az container attach: Attach local standard output and error streams to a container in a
+        container group.
+
+    Arguments
+        --container-name   : The container to attach to. If omitted, the first container in the
```
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/README.rst]: No. Container name is not a part of ARM id.

File: src/command_modules/azure-cli-container/README.rst [hunk 5]
```diff
@@ -177,3 +178,30 @@ Commands to list Azure container groups by resource group
...
+                            container group will be chosen.
+
+    Resource Id Arguments
+        --ids              : One or more resource IDs (space delimited). If provided, no other 'Resource
+                            Id' arguments should be specified.
+        --name -n          : The name of the container group.
```
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/README.rst]: We are already using this pattern for `az container logs` where `-n` is container group name and `--container-name` is the container name.  Should I change both?

File: src/command_modules/azure-cli-container/setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/setup.py]: Actually I'm not using any color function from the package. I'm just using it for enable the ASCII encoding in Windows.

[the remark this answers is not in the document]
Comment (derekbekoe) [reply src/command_modules/azure-cli-container/setup.py]: Can you remove the explicit version and just put `colorama`?
Would rather the version constraint was kept in a single place.
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/setup.py#L56

File: src/command_modules/azure-cli-container/README.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (tjprescott) [reply src/command_modules/azure-cli-container/README.rst]: What is a container group and where do you create them?

[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/README.rst]: A container group is a ARM resource. Can be created by command `az container create`

File: src/command_modules/azure-cli-container/setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/setup.py]: @derekbekoe Will do

File: src/command_modules/azure-cli-container/README.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (yolo3301) [reply src/command_modules/azure-cli-container/README.rst]: A container is composed of containers. In today CLI, creating a container group basically equals creating a container group with only one container in it.

**Comment by yolo3301** [comment]
Ping @tjprescott

**Pull request merged**